docs/test_instructions_per_dependency_update.md

```diff
@@ -0,0 +1,859 @@
+document up-to-date a low-effort and minimally maintenance-intensive task.
+
+### Should this document be blindly trusted?
+
+No engineer should blindly trust and strictly adhere to the testing instructions provided in this
+document. These instructions serve as a starting point and a reminder for engineers working on a
+dependency update to ensure they test the core screens. Engineers should exercise their best
+judgment and conduct additional testing if they deem it necessary.
+
+### Do we need testing instructions for every dependency update?
+
+Some dependency updates are quite broad and apply to the entire app, making it challenging to
+provide specific testing instructions (e.g., `fragment`, `appcompat`, `compose`, etc.). In such
+cases, engineers should use their best judgment to perform a smoke test on the entire app to ensure
+its correctness. Engineers should rely on their intuition to identify which dependencies warrant
+testing instructions. Therefore, this document should be regarded as a set of helpful guidelines
+rather than strict requirements.
+
+## Content
+
+1. Plugin
+    1. [Navigation](#navigation)
+2. AndroidX Jetpack
+    1. [WorkManager](#workmanager)
+3. AndroidX/Google
+    1. [Preference](#preference)
+    2. [ExoPlayer](#exoplayer)
+    3. [Webkit](#webkit)
+4. Firebase/Google
+    1. [Firebase](#firebase)
+    2. [FirebaseIid](#firebaseiid)
+    3. [MLKitBarcodeScanning](#mlkitbarcodescanning)
+    4. [PlayServicesAuth](#playservicesauth)
+    5. [PlayServicesCoreScanner](#playservicescodescanner)
+5. Network
+    1. [Okio](#okio)
+6. Tool
+    1. [Zendesk](#zendesk)
+    2. [JSoup](#jsoup)
+7. Other Core
+    1. [AutoService](#autoservice)
+    2. [KotlinPoet](#kotlinpoet)
+8. Other UI
+    1. [Lottie](#lottie)
+    2. [UCrop](#ucrop)
+9. [Smoke Test](#smoke-test)
+10. [Special](#special)
+
+ℹ️ Every test instruction should be prefixed with one of the following:
+- [JP/WP] This test applies to both, the `Jetpack` and `WordPress` apps.
+- [JP] This test applies to the `Jetpack` app only.
+- [WP] This test applies to the `WordPress` app only.
+
+-----
+
+### Navigation [[navigationVersion](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/settings.gradle#L1)] <a name="navigation"></a>
+
+<details>
+    <summary>1. [JP/WP] Image Editing Flow [libs:image-editor]</summary>
+
+- Add a new `blog` post.
+- Add a new `image` block.
+- Choose an image and wait for it to be uploaded within the `image` block.
+- Click on the `media options` of this image (top right) and then click `edit`.
+- Verify that the `Edit Image` screen is shown and functioning as expected.
+- Crop the image and click the `done` menu option (top right).
+- Make sure the image is updated accordingly.
+
+</details>
+
+-----
+
+### WorkManager [[androidxWorkManagerVersion](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/build.gradle#L54)] <a name="workmanager"></a>
```

Comment:
> All other links point to the first line (#L1), but the androidxWorkManagerVersion and androidxPreferenceVersion links point to #L54 and #L50.

Spot on! 💯 

> I believe you avoided stating line numbers in case they change in the future.

Exactly! 🎯 

> We can remove all line numbers from the links and just link the file.

True, will do exactly that! 🥇

Reply:
Done: a6f1cd00b94de220c46607e8897ed8159e850e9b + 847d28b32c5632bd954ea96f0ddef35811c71530